astropy/table/table.py

```diff
@@ -1795,7 +1795,15 @@ def show_in_notebook(self, *, backend="ipydatagrid", **kwargs):
 
         """
         if backend == "ipydatagrid":
-            from astropy.table.notebook_backends import ipydatagrid
+            try:
+                import pandas  # noqa: F401
+
+                from astropy.table.notebook_backends import ipydatagrid
+            except ImportError:
+                raise ImportError(
+                    "The default option for show_in_notebook now requires pandas "
+                    "and ipydatagrid to also be installed"
```

Comment:
```suggestion
                    "and ipydatagrid to also be installed, or please consider using the astropy[jupyter] extras"
```

Reply:
Hmm. But `pandas` is under `all`, so should the text be `astropy[all,jupyter]`?

Reply:
If this is a feature that I need for the notebooks then pandas should be in here, too. (And then [jupyter] be included in [all]). Basically the logic should be that the end user shouldn't need to install any of these one by one but by choosing the appropriate extras group(s), but still not being forced to install everything.

Reply:
That was done in https://github.com/astropy/astropy/pull/16340 by @nstarman . I did not follow the discussions closely. Reordering pyproject.toml might be out of scope if the ask is only for a clearer error message.

Reply:
Well, the ask is also to provide an extras that notebooks need, not adding pandas alongside with pydatagrid was imo an oversight.